01.md

```diff
@@ -35,6 +35,10 @@ Once `LNURL` is decoded:
 
 Neither status codes or any HTTP Header has any meaning. Servers may use whatever they want. Clients should ignore them (and be careful when using libraries that treat responses differently based on headers and status codes) and just parse the response body as JSON, then interpret it accordingly.
 
+## CORS Browser Compatibility
+
+Reverse proxies or application endpoints must use the CORS Header `Access-Control-Allow-Origin: *` for LNURL's to work with browser-based wallets, services and other webapps. 
```

Comment:
```suggestion
Application endpoints that need to be accessed when using the LNUrl must set the header `Access-Control-Allow-Origin: *` for LNURL's to work with browser-based wallets, services and other webapps. They also must not set the `Access-Control-Allow-Methods` header in such a way that `GET` requests are denied.
```

Reply:
if add `GET`, could be useful mention `OPTIONS` as well

Reply:
> if add `GET`, could be useful mention `OPTIONS` as well

Options requests can not be denied with a header, as they are a browser feature. However, servers should respond to options request. I'll edit my suggestion.

Reply:
how about replacing that entire paragraph with "Requests must support CORS"?

Reply:
Maybe `Application endpoints that need to be accessed when interacting with an LNURL must set CORS headers in such a way that they are reachable from any web origin.` would be better?

Reply:
Yes.

I just think we shouldn't describe what CORS is.

Reply:
These things are typically done at reverse proxies, people know what nginx and caddy are in their stack. To keep the explanation simple list the header snippets.

Reply:
Setting the header in a proxy is a workaround, not a solution.

Reply:
What? Its for the service implementing the specs... the service which runs behind a reverse proxy. Who's directly exposing their app? nginx or caddy are used effectively everywhere, and a reverse proxies directive is part of an application stack. They are not separate.